### PR TITLE
ci: continuing linting when one kind errors

### PR DESCRIPTION
**Pull request by turadg**
## Description

It's helpful to developers that when CI fails you are informed what you need to do to make it pass. Presently when one kind of lint check fails, the logic fails fast and you get no information about subsequent ones. This was due to the `steps` in Github workflow and the `&&` in the `yarn lint` scripts.

This makes the `yarn lint` command check all the defined `lint:foo` commands and continue when any error. It accomplishes this using https://github.com/mysticatea/npm-run-all . 

It also:
- update CI to use the one `yarn lint` so its steps don't fail fast
- remove `lint-check` command since that's what `lint` does (checks)
- removes `tsc` from `lint-fix` since it can't do any fixing.
- remove the commands that used `.eslintrc-jessie.js` because that files doesn't exist (superceded by `plugin:@jessie.js/recommended`)

### Security Considerations

Introduction of a new package. It's only a devDependency and is maintained by a well-regarded project lead who's a member of ESLint, Babel, and Vue.js.

### Documentation Considerations

Notify developers to that `yarn lint-check` no longer works and to use `yarn lint` instead. 

### Testing Considerations

Locally tested that the different jobs run and failure of any is conveyed for the lint overall.

CI passes.

### CODE REVIEW

**Comment by dckc** [comment]
> The current `lint` check in CI ... conflates type checking with linting

conflates? or usefully combines? they're both incomplete static checks on code, yes?

I suggest surveying more of the team, but to my mind, it's a feature, not a bug, that `yarn lint` does both.

If you want separate `lint:eslint` and `lint:types` ci jobs, very well. But I'm not convinced there's a good reason to change what `yarn lint` means.

**Comment by mhofman** [comment]
I'm not sure about this. I'm actually more of the opinion of use "lint" as an umbrella not restricted to "eslint", but also for types and even an explicit "prettier" check, so I actually prefer the conflation.

As to the `&&` making these sequential, it sure would be nice to have all output available when multiple errors exist, but is it really necessary?

**Comment by turadg** [comment]
> conflates? or usefully combines? they're both incomplete static checks on code, yes?

Well taken. I was thinking of "lint" as "eslint" but that's just one type of lint.

> As to the && making these sequential, it sure would be nice to have all output available when multiple errors exist, but is it really necessary?

I think it's a good requirement that when CI fails you are informed what you need to do to make it pass. Presently when one kind of lint check fails, you get no information about ones later in the `&&` expression.

So that means:
> ~conflates type checking with linting~
> the && expression means if any typechecks fail, linting isn't run

Instead I'll solve the latter so that the new CI check isn't necessary. (Saves setup time too.)

File: README.md [hunk 1]
```diff
@@ -65,7 +65,7 @@ constraints). Our goal is to remove all the unique-to-a-subproject deps.
 
 When one subproject depends upon another, `node_modules/` will contain a
 symlink to the subproject (e.g. `ERTP` depends upon `marshal`, so
-`node_modules/@agoric/marshal` is a symlink to `packages/marshal`).
+`node_modules/@endo/marshal` is a symlink to `packages/marshal`).
```
Comment (dckc) [inline README.md]: I doubt the symlink thing applies across monorepos. So maybe change the example from ERTP->marshal to Zoe->ERTP.

Comment (turadg) [reply README.md]: Dunno how that change got in the PR. Gone now.

**Review by dckc — APPROVED** [review]
I'm a little hesitant to speak for the team on this... especially without an issue. But I guess I'm OK to give this my usual level of approval, which is: if it turns out to be broken, I agree to be responsible to fix it.

File: package.json [hunk 2]
```diff
@@ -81,10 +82,10 @@
     "link-cli": "yarn run create-agoric-cli",
     "create-agoric-cli": "node ./scripts/create-agoric-cli.cjs",
     "format": "yarn prettier --write 'packages/**/*.{js,jsx,ts,tsx}' 'golang/**/*.{js,jsx,ts,tsx}'",
```
Comment (mhofman) [inline package.json:84]: Should we rename to `lint-fix:format` in the principle of consistency? And introduce a `lint-fix` that performs the above and a workspace `lint-fix` (for the eslint fix)

File: packages/bundle-source/package.json [hunk 3]
```diff
@@ -10,9 +10,6 @@
     "test:c8": "c8 $C8_OPTIONS ava --config=ava-nesm.config.js",
     "test:xs": "exit 0",
     "lint-fix": "eslint --fix '**/*.js'",
-    "lint-check": "yarn lint",
-    "lint-fix-jessie": "eslint -c '.eslintrc-jessie.js' --fix '**/*.js'",
-    "lint-check-jessie": "eslint -c '.eslintrc-jessie.js' '**/*.js'",
```
Comment (mhofman) [inline packages/bundle-source/package.json:15]: Where did these go?

Comment (turadg) [reply packages/bundle-source/package.json:15]: I'll mention in the notes: there is no `.eslintrc-jessie.js` so these weren't even working. We have `"plugin:@jessie.js/recommended"` in eslint-config so it's handled by that.

File: package.json (hunk 2 above)
Comment (turadg) [reply package.json:84]: I've been burned by `eslint --fix` enough not to trust it so I'd personally shy away from using a catch-all `yarn lint-fix` but I wouldn't object to a PR making that change.

File: packages/bundle-source/package.json (hunk 3 above)
Comment (mhofman) [reply packages/bundle-source/package.json:15]: Ping @michaelfig I assume?

File: package.json (hunk 2 above)
Comment (mhofman) [reply package.json:84]: Agreed! But I believe it's used by some. I personally relies on my editor for formatting auto fixes, so I use neither. I'll let others chime in.

File: packages/bundle-source/package.json (hunk 3 above)
Comment (dckc) [reply packages/bundle-source/package.json:15]: yes, `plugin:@jessie.js/recommended` along with `@jessie-check` is the current mechanism.

**Pull request merged**